### ISSUE TITLE
Bug: no-dupe-class-members reporting "Duplicate name ''" (duplicate name is blank!)

### ISSUE THREAD

**Issue by mwittmann** [eslint/eslint#15232]
### Environment

Node version: v17.0.1
npm version: v8.1.0
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found
Operating System: Windows 10 Professional

### What parser are you using?

@babel/eslint-parser

### What did you do?

I am getting unexpected `error  Duplicate name ''  no-dupe-class-members`. Yes, the `name` is blank... nothing between the two apostrophes `''`. Although I'm not a javascript expert, I don't see anything wrong with the class definition.

FWIW, I'm running this with `pnpm v.6.19.1`. Command line is `pnpm exec eslint .\src\js\nodupe-error-demo.js`

<details>
<summary>Configuration</summary>
```json
{
  "root": true,
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "es6": true,
    "jquery": true,
    "jasmine": true,
    "amd": true
  },
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": "./webpack/webpack.config.epic.js"
      }
    }
  },
  "ignorePatterns": [
    "mdb"
  ],
  "extends": [
    "airbnb-base",
    "plugin:prettier/recommended"
  ],
  "parserOptions": {
    "ecmaVersion": 8,
    "sourceType": "module"
  },
  "rules": {
    "no-await-in-loop": "error",
    "no-compare-neg-zero": "error",
    "no-extra-parens": "error",
    "no-prototype-builtins": "off",
    "no-template-curly-in-string": "error",
    "valid-jsdoc": "error",
    "accessor-pairs": "error",
    "array-callback-return": "error",
    "block-scoped-var": "error",
    "class-methods-use-this": "off",
    "complexity": "off",
    "consistent-return": "error",
    "curly": "error",
    "default-case": "error",
    "dot-location": [
      "error",
      "property"
    ],
    "dot-notation": "error",
    "eqeqeq": "error",
    "guard-for-in": "error",
    "no-alert": "error",
    "no-caller": "error",
    "no-div-regex": "error",
    "no-else-return": "error",
    "no-empty-function": "error",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": "error",
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-invalid-this": "off",
    "no-iterator": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-loop-func": "error",
    "no-multi-spaces": [
      "error",
      {
        "exceptions": {
          "AssignmentExpression": true,
          "ArrowFunctionExpression": true,
          "CallExpression": true,
          "VariableDeclarator": true
        }
      }
    ],
    "no-multi-str": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "off",
    "no-proto": "error",
    "no-restricted-properties": "error",
    "no-return-assign": "off",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-throw-literal": "error",
    "no-unmodified-loop-condition": "error",
    "no-unused-expressions": "error",
    "no-useless-call": "error",
    "no-useless-concat": "error",
    "no-useless-escape": "error",
    "no-useless-return": "off",
    "no-void": "error",
    "no-warning-comments": "off",
    "no-with": "error",
    "prefer-promise-reject-errors": "error",
    "radix": "error",
    "require-await": "error",
    "vars-on-top": "error",
    "wrap-iife": [ "error", "inside" ], // changed default ("outside")
    "yoda": "error",
    "strict": "error",
    "init-declarations": "off",
    "no-catch-shadow": "error",
    "no-label-var": "error",
    "no-restricted-globals": "error",
    "no-shadow": "off",
    "no-shadow-restricted-names": "error",
    "no-undef-init": "error",
    "no-undefined": "off",
    "no-use-before-define": "off",
    "callback-return": "off",
    "global-require": "error",
    "handle-callback-err": "error",
    "no-mixed-requires": "error",
    "no-new-require": "error",
    "no-path-concat": "error",
    "no-process-env": "error",
    "no-process-exit": "error",
    "no-restricted-modules": "error",
    "no-sync": "error",
    "array-bracket-spacing": "error",
    "block-spacing": "error",
    "brace-style": "error",
    "camelcase": "error",
    "capitalized-comments": "off",
    "comma-dangle": "off",
    "comma-spacing": "error",
    "comma-style": "error",
    "computed-property-spacing": "error",
    "consistent-this": "error",
    "eol-last": "error",
    "func-call-spacing": "error",
    "func-name-matching": "error",
    "func-names": "off",
    "func-style": [
      "error",
      "declaration",
      {
        "allowArrowFunctions": true
      }
    ],
    "id-blacklist": "error",
    "id-length": "off",
    "id-match": "error",
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1
      }
    ],
    "jsx-quotes": "error",
    "key-spacing": "off",
    "keyword-spacing": "error",
    "line-comment-position": "off",
    "lines-around-comment": "off",
    "lines-around-directive": "error",
    "max-depth": [
      "error",
      10
    ],
    "max-len": "off",
    "max-lines": "off",
    "max-nested-callbacks": "error",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "error",
    "multiline-ternary": "off",
    "new-cap": [
      "error",
      {
        "capIsNewExceptionPattern": "$.*"
      }
    ],
    "new-parens": "error",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": [
      "error",
      {
        "ignoreChainWithDepth": 5
      }
    ],
    "no-array-constructor": "error",
    "no-bitwise": "error",
    "no-continue": "off",
    "no-inline-comments": "off",
    "no-lonely-if": "error",
    "no-mixed-operators": "off",
    "no-multi-assign": "error",
    "no-multiple-empty-lines": "error",
    "no-negated-condition": "off",
    "no-nested-ternary": "error",
    "no-new-object": "error",
    "no-plusplus": "off",
    "no-restricted-syntax": "error",
    "no-tabs": "error",
    "no-ternary": "off",
    "no-trailing-spaces": "error",
    "no-underscore-dangle": "off",
    "no-unneeded-ternary": "error",
    "no-whitespace-before-property": "error",
    "nonblock-statement-body-position": "error",
    "object-curly-newline": [
      "error",
      {
        "consistent": true,
        "multiline": true
      }
    ],
    "object-curly-spacing": [
      "error",
      "always"
    ],
    "object-property-newline": "error",
    "one-var": [
      "error",
      "never"
    ],
    "one-var-declaration-per-line": "error",
    "operator-assignment": "error",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "quote-props": [
      "error",
      "as-needed"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "require-jsdoc": "off",
    "semi": [
      "off"
    ],
    "semi-spacing": "error",
    "sort-keys": "off",
    "sort-vars": "error",
    "space-before-blocks": "error",
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "named": "never"
      }
    ],
    "space-in-parens": "error",
    "space-infix-ops": "error",
    "space-unary-ops": "error",
    //"spaced-comment": "error",
    "spaced-comment": "off",
    "template-tag-spacing": "error",
    "unicode-bom": "error",
    "wrap-regex": "off",
    "arrow-body-style": "off",
    "arrow-spacing": "error",
    "generator-star-spacing": "error",
    "no-confusing-arrow": "error",
    "no-duplicate-imports": "error",
    "no-restricted-imports": "error",
    "no-useless-computed-key": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-var": "error",
    "object-shorthand": "error",
    "prefer-arrow-callback": "error",
    "prefer-const": "error",
    "prefer-destructuring": "off",
    "prefer-numeric-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "error",
    "rest-spread-spacing": "error",
    // "sort-imports": "error",
    "symbol-description": "error",
    "template-curly-spacing": "error",
    "yield-star-spacing": "error",
    "import/prefer-default-export": "off"
  }
}
```
</details>

```js
class SampleClass {
  #_originalArray;

  #_currentArray;

  #_originalString;

  #_currentString;

  constructor(myArray) {
    this.#init(myArray);
  }

  ////////////////////////////
  // public
  get originalString() {
    return this.#_originalString;
  }

  get currentString() {
    return this.#_currentString;
  }

  get originalArray() {
    return this.#_originalString;
  }

  get currentArray() {
    return this.#_currentString;
  }

  set currentArray(newArray) {
    this.#currentArray = newArray;
  }

  ////////////////////////////
  // private getters and setters
  // keep string versions in sync w/ array versions
  set #originalArray(myArray) {
    this.#_originalArray = myArray;
    this.#_originalString = this.#originalArray.join('|');
  }

  // YIELDS: error  Duplicate name ''  no-dupe-class-members
  set #currentArray(myArray) {
    this.#_currentArray = myArray;
    this.#_currentString = this.#currentArray.join('|');
  }

  get #originalArray() {
    return this.#_originalArray;
  }

  // YIELDS: error  Duplicate name ''  no-dupe-class-members
  get #currentArray() {
    return this.#_currentArray;
  }

  // YIELDS: error  Duplicate name ''  no-dupe-class-members
  #init(myArray) {
    // save original selections
    this.#originalArray = myArray;
    this.#currentArray = this.#originalArray;
  }
}

const oSample = new SampleClass(['a', 'b', 'c']);
oSample.currentArray = ['d', 'e', 'f'];
```
Source: [nodupe-error-demo.zip](https://github.com/eslint/eslint/files/7445359/nodupe-error-demo.zip)

### What did you expect to happen?

The javascript class should parse without `eslint` errors.

### What actually happened?

`eslint` returned the following errors (also marked in the .js source):
```
  45:3  error  Duplicate name ''  no-dupe-class-members
  55:3  error  Duplicate name ''  no-dupe-class-members
  60:3  error  Duplicate name ''  no-dupe-class-members

✖ 3 problems (3 errors, 0 warnings)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ljharb** [comment]
I think class fields aren't supported until eslint 8.

**Comment by nzakas** [comment]
That is correct. Because you are using Babel-eslint, you are able to parse class fields but the ESLint rules in v6.8.0 do not support class fields and so won’t work correctly. You’ll need to upgrade to ESLint 8 to fix this.

**Comment by mwittmann** [comment]
Got it. Thanks so much!
